configure.ac

```diff
@@ -507,8 +507,11 @@ then
       [],
       [if test "x$with_libacl" != xcheck; then AC_MSG_ERROR(Cannot find libacl library); fi])
     AC_CHECK_HEADERS([acl.h sys/acl.h acl/libacl.h],
-      [libacl_header_found=yes],
-      [if test "x$with_libacl" != xcheck; then AC_MSG_ERROR(Cannot find libacl header files); fi])
+      [libacl_header_found=yes])
+    if test "x$libacl_header_found" != "xyes" && test "x$with_libacl" != xcheck;
```

Comment:
You move the check outside the AC_CHECK_HEADERS macro, which means that the check will likely ignore the two first headers, only checking the last one, because `libacl_header_found` is set 3 times, once for each header file.

Can you explain what the issue is exactly so that we figure a better way to fix it? What error are you seeing? Which header files are present on your system?

Reply:
> You move the check outside the AC_CHECK_HEADERS macro

This is how it was done prior to 41c0afd ("Improve readability"), I'm just reverting that portion of the commit.  It seems pretty clear that commit did not intend to change the semantics of that check, so this is a regression.

> Can you explain what the issue is exactly so that we figure a better way to fix it? What error are you seeing?
```
$ ./configure --with-libacl
[SNIP]
checking for acl_init in -lacl... yes
checking acl.h usabilty... no
checking acl.h presense... no
checking for acl.h... no
configure: error: Cannot find libacl header files
```
> Which header files are present on your system?
```
ls /usr/include/acl.h /usr/include/sys/acl.h /usr/include/acl/libacl.h
/usr/include/sys/acl.h
/usr/include/acl/libacl.h
```